test/specs/commands/create_account.js

```diff
@@ -20,30 +20,96 @@ import * as then from '../../steps/3_then';
 
 describe('create account command', () => {
 	beforeEach(setUpCommandCreateAccount);
+	Given(
+		'a passphrase "minute omit local rare sword knee banner pair rib museum shadow juice" with private key "314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and public key "7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and address "2167422481642255385L"',
+		given.aPassphraseWithPrivateKeyAndPublicKeyAndAddress,
+		() => {
+			Given(
+				'the passphrase is generated by the createMnemonicPassphrase function',
+				given.thePassphraseIsGeneratedByTheCreateMnemonicPassphraseFunction,
+				() => {
+					Given('an action "create account"', given.anAction, () => {
+						When('the action is called', when.theActionIsCalled, () => {
+							Then(
+								'it should resolve to an array having one object with the passphrase, the private key, the public key and the address',
+								then.itShouldResolveToAnObjectWithThePassphraseThePrivateKeyThePublicKeyAndTheAddress,
+							);
+						});
+					});
+				},
+			);
+		},
+	);
 	Given(
 		'a crypto instance has been initialised',
 		given.aCryptoInstanceHasBeenInitialised,
 		() => {
 			Given(
-				'a passphrase "minute omit local rare sword knee banner pair rib museum shadow juice" with private key "314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and public key "7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and address "2167422481642255385L"',
-				given.aPassphraseWithPrivateKeyAndPublicKeyAndAddress,
+				'a valid integer corresponding to number of accounts "1"',
```

Comment:
It would be good to have tests for larger numbers too, which was after all the point of the feature.

Reply:
@willclarktech , I was thinking about how to write the tests for large numbers.

As far as I understand the code,
we make the functions return the passphrase, keys and address by doing:
lisk.cryptography.getAddressFromPublicKey.returns(address);

Here if we think that we can create cases where we have a multiple number of accounts, how do we change the test?
We can have something like this for 2 accounts:

``` 
Given(
'a valid integer corresponding to number of accounts "2" 
 for the first account, passphrase "smart couch position escape begin retreat summer snack fruit virtual climb shine" with private key "3818764e457d57dada65de626660039307fdd3fbf83743095579188669a110a8f4175ac1892a450df668b75a90e7a3b5066d0bc90ee91a808de686a8867dc46d" and public key "f4175ac1892a450df668b75a90e7a3b5066d0bc90ee91a808de686a8867dc46d" and address "7195654441783367976L"
  and for second account,  passphrase "minute omit local rare sword knee banner pair rib museum shadow juice" with private key "314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and public key "7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588" and address "2167422481642255385L"
'),
```	
The parsing logic can be written here, but my doubt is that how do we change `aPassphraseWithPrivateKeyAndPublicKeyAndAddress` function to return different parameters in each iteration of create account call.

Reply:
My suggestion would be to have distinct steps for distinct setup requirements. So one step would specify that the number of accounts is 2, another step would specify the passphrase with corresponding keys and address, and a third step would specify another passphrase/keys/address set: `anotherPassphraseWithPrivateKeyAndPublicKeyAndAddress`.

Reply:
For distinct variables stored in the test context we need distinct names. So e.g. `passphrase` and `passphrase2`, `publicKey` and `publicKey2` etc.